plaid/item.go

```diff
@@ -277,7 +292,14 @@ func (c *Client) CreatePublicToken(accessToken string) (resp CreatePublicTokenRe
 }
 
 // CreateItemAddToken generates a token which is used to initialize Link.
+//
+// You can optionally supply identity fields you know, and may have verified,
+// for the user. This will allow us to optimise their experience if we have seen
+// them before.
+//
+// Beta: this endpoint is still in beta.
 func (c *Client) CreateItemAddToken(userFields ItemAddTokenUserFields) (resp CreateItemAddTokenResponse, err error) {
+	prepareUserFieldsForSend(&userFields)
```

Comment:
(nit) instead of having the unused bool fields on the json request struct and mutating fields here, as an alternative you could do something like:

```go
type VerificationTime struct {
  Time *time.Time
  IsVerified bool
}

func (v VerficiationTime) MarshalJSON() ([]byte, error) {
  if v.Time == nil && v.isVerified {
    return json.Marshal(verificationDateUnknown)
  }
  return json.Marshal(v.Time)
}

type ItemAddTokenUserFields struct {
  EmailAddressVerifiedTime VerificationTime `json:"email_address_verified_time"`
}
```

that way all the field processing is contained within the struct and we don't rely on `CreateItemAddToken` to get the right json serialization

Reply:
Like it, very elegant! Have refactored towards that pattern. I dropped the `IsVerified` field as I think a pointer is clearer.